### PR TITLE
[EDT-1386] Add methods to create barcode and set basic settings

### PR DESCRIPTION
**Pull request by NicolaVerbeeck**
This PR

## PR Guidelines

- [x] I have read the [contribution and PR guidelines](/chili-publish/studio-sdk/blob/main/CONTRIBUTING.md)

## Related tickets

-   [EDT-1386](https://chilipublishintranet.atlassian.net/browse/EDT-1386)

## Screenshots

[EDT-1386]: https://chilipublishintranet.atlassian.net/browse/EDT-1386?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/sdk/src/controllers/FrameController.ts [hunk 1]
```diff
@@ -383,6 +384,23 @@ export class FrameController {
...
+     * not specified, that property will use a sensible default
+     *
+     * @param type the type of barcode to create
+     * @param x optional X coordinate of the new frame within the layout
+     * @param y optional Y coordinate of the new frame within the layout
+     * @param width optional Width of the new frame within the layout
+     * @param height optional Height of the new frame within the layout
+     * @returns
+     */
+    createBarcodeFrame = async (type: BarcodeType, x?: number, y?: number, width?: number, height?: number) => {
```
Comment (Matthiee) [inline packages/sdk/src/controllers/FrameController.ts:399]: Given that all these properties are optional I would suggest the following:

```suggestion
    createBarcodeFrame = async (type: BarcodeType, position?: { x?: number, y?: number, width?: number, height?: number }) => {
```

File: packages/sdk/src/types/BarcodeTypes.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { ColorUsage } from './ColorStyleTypes';
+
+export enum BarcodeType {
```
Comment (Matthiee) [inline packages/sdk/src/types/BarcodeTypes.ts:3]: These probably need to be exported in `index.ts`?

**Review by Matthiee — CHANGES_REQUESTED** [review]

**Comment by Dvergar** [comment]
Just a side-thought, since `Barcode` has its own controller maybe it would be great to explode frame creation and frame specific helpers into their own controllers as well for `Shape` and `Text` frames for consistency's sake (that would be easily retro-compatible as well).

File: packages/sdk/src/controllers/FrameController.ts [hunk 3]
```diff
@@ -383,6 +384,25 @@ export class FrameController {
...
+     * @param position optional position object where you can specify the x, y, width and height of the barcode frame
+     * @returns
+     */
+    createBarcodeFrame = async (
+        type: BarcodeType,
+        position?: { x?: number; y?: number; width?: number; height?: number },
+    ) => {
+        const res = await this.#editorAPI;
+        return res
+            .addBarcodeFrame(type, position?.x, position?.y, position?.width, position?.height)
```
[the remark this answers is not in the document]
Comment (NicolaVerbeeck) [reply packages/sdk/src/controllers/FrameController.ts:402]: In the implementation for addFrame on engine side we allready have to do an exhaustive search to see if we are actually adding a 'regular' frame or a 'shape' frame. This locks the implementation down to a clearer naming scheme

**Comment by NicolaVerbeeck** [comment]
> Just a side-thought for a future refactoring: since `Barcode` has its own controller maybe it would be great to explode frame creation and frame specific helpers into their own controllers as well for `Shape` and `Text` frames for consistency's sake (that would be easily retro-compatible as well).

We discussed this with @brapoprod but decided to move this to v2 instead to keep things consistent

File: packages/sdk/src/controllers/LayoutController.ts [hunk 4]
```diff
@@ -229,6 +229,18 @@ export class LayoutController {
         return res.setLayoutFillColor(id, JSON.stringify(color)).then((result) => getEditorResponseData<null>(result));
     };
 
+    /**
+     * This is a convenience method to enable or disable the fill color of a specific layout.
+     * Note: Depending on the layout intent, disabling might not be valid (eg disabling for digitalAnimated)
+     *
+     * @param id the id of a specific layout
+     * @param enabled  whether the fill color should be enabled or disabled
```
Comment (Dvergar) [inline packages/sdk/src/controllers/LayoutController.ts:237]:
```suggestion
     * @param enabled whether the fill color should be enabled or disabled
```
🔎

File: packages/sdk/src/controllers/BarcodeController.ts [hunk 5]
```diff
@@ -96,12 +96,24 @@ export class BarcodeController {
      */
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     getBarcodeConfigationOptions = (type: BarcodeType): BarcodeConfigurationOptions => {
+        var allowToggleText = true;
```
Comment (Dvergar) [inline packages/sdk/src/controllers/BarcodeController.ts:99]:
```suggestion
        let allowToggleText = true;
```

**Review by Dvergar — CHANGES_REQUESTED** [review]

**Review by Dvergar — DISMISSED** [review]

**Review by psamusev — APPROVED** [review]

**Review by srares76 — DISMISSED** [review]

**Review by brapoprod — APPROVED** [review]

**Review by Matthiee — APPROVED** [review]

**Review by srares76 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -1,3 +1,3 @@
 *       @brapoprod @abdelhalimkhouas @alexandraFlavia9 @pietervp @psamusev @srares76
 /packages/actions/ @Matthiee
-
+/packages/sdk/ @Matthiee @Dvergar @NicolaVerbeeck @olegsolodukhin @evan-mcgeek @brapoprod
```
Comment (Matthiee) [inline .github/CODEOWNERS:3]:
```suggestion
/packages/sdk/ @chili-publish/flutter
```

**Review by Dvergar — DISMISSED** [review]

**Review by srares76 — APPROVED** [review]

**Review by brapoprod — APPROVED** [review]

**Pull request merged**